modules/gapi/src/api/gmat.cpp

```diff
@@ -33,22 +33,47 @@ const cv::GOrigin& cv::GMat::priv() const
     return *m_priv;
 }
 
+template <typename T> std::vector<cv::GMatDesc> vec_descr_of(const std::vector<T> &vec)
+{
+    std::vector<cv::GMatDesc> vec_descr;
```

Comment:
IMHO calling `vec_descr.reserve(vec.size())` will minimize number of memory allocations